GridNav custom layers system

K1llMan opened the pull request.

K1llMan committed:
GridNav custom layers system

K1llMan committed:
Fix initial grid checking

K1llMan commented:
![fullscreengrid](https://cloud.githubusercontent.com/assets/13403439/12377410/56263bf2-bd2e-11e5-8d22-7cef6f642767.jpg)
Full screen grid testing. Need to crop by visible terrain.

K1llMan commented:
![fow_crop](https://cloud.githubusercontent.com/assets/13403439/12403171/4af23e6e-be43-11e5-93f5-3537200573ae.jpg)
Crop by fow. Is there any way to get correct height of the map coord?

MNoya commented:
Make sure your particle has cullradius -1

![img](http://puu.sh/mARyo/72a764509a.png)

K1llMan commented:
![grid_units](https://cloud.githubusercontent.com/assets/13403439/12430802/d253d2e0-bf02-11e5-8d64-122ccb10fc08.jpg)
Grid with unit checking.

K1llMan committed:
Full screen grid

Full screen grid with unit checking and cropping by fow. Still need some
perfomance tweaks.

K1llMan commented:
![fullgrid](https://cloud.githubusercontent.com/assets/13403439/12462726/e744456c-bfd0-11e5-8f28-44ed301bed12.jpg)

K1llMan committed:
Grid performance tweak, grid colors

K1llMan committed:
GNV callbacks

heme3ic commented:
там выше писали про "не будет ли гриднав выдавать игроков", но кому? петрам этот гриднав вообще не нужен, а если зрителей "выдавать" зрителям, то ничего плохого не будет

K1llMan committed:
Grid modes, show enemy on grid as blocked

K1llMan committed:
StarCraft-like grid, new grid mode, alt-press fix

K1llMan committed:
data() fix

K1llMan committed:
Split GridNav and BuildingHelper modules

K1llMan committed:
Disable visible grid, some styles fix

K1llMan committed:
Custom users config for GNV

K1llMan committed:
Grid optimization, in-game grid settings menu

K1llMan committed:
ColorPicker UI-element, grid custom colors

K1llMan committed:
Changed min player count to random shuffle

K1llMan committed:
Added default config request

K1llMan committed:
Shuffle messages

K1llMan committed:
Prefer list print

K1llMan committed:
!111

K1llMan committed:
Changed value

K1llMan committed:
Remove debug prints

K1llMan committed:
Index fix

K1llMan committed:
Merge

K1llMan committed:
Camera_test

K1llMan committed:
Update skills info for enemy and spectators
